src/main/java/impl/DataProcessorImpl.java

```diff
@@ -0,0 +1,24 @@
+package impl;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.DataProcessor;
+import strategy.OperationHandler;
+import strategy.OperationStrategy;
+
+public class DataProcessorImpl implements DataProcessor {
+    private OperationStrategy operationStrategy;
+
+    public DataProcessorImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        for (FruitTransaction transaction : fruitTransactions) {
+            OperationHandler operationHandler = operationStrategy
+                    .getHandler(transaction.getOperation());
+            operationHandler.apply(transaction);
+        }
```

Comment:
Ensure that all services are independent and are called in the right order in the `main()` method. The processing logic should ideally be moved to the `main()` method where you can call services step by step, passing the result of one service to the next.